docs/usage/PRODUCT_SYNC.md

```diff
@@ -21,51 +34,132 @@ against a [ProductDraft](https://docs.commercetools.com/http-api-projects-produc
-     ````java
-     final List<ProductDraft> productDrafts = ProductReferenceResolutionUtils.mapToProductDrafts(products);
-     ````
-     > Note: Some references in the product like `state`, `customerGroup` of prices, and variant attributes with type `reference` do not support the `ResourceIdentifier` yet, 
-      for those references you need to provide the `key` value on the `id` field of the reference. This means that calling `getId()` on the
-      reference would return its `key`. 
-    
-    - For resolving `key-value-document` (custom object) references on attributes of type `Reference`, `Set` of `Reference`, `NestedType` or `Set` of `NestedType`, The `id` field of the reference in the attribute draft should be defined in the correct format. 
-    The correct format must have a vertical bar `|` character between the values of the container and key.
-    For example, if the custom object has a container value `container` and key value `key`, the `id` field should be `container|key"`,  
-    also, the key and container value should match the pattern `[-_~.a-zA-Z0-9]+`.
+````java
+final SphereClientConfig clientConfig = SphereClientConfig.of("project-key", "client-id", "client-secret");
+
+final SphereClient sphereClient = ClientConfigurationUtils.createClient(clientConfig);
+````
+
+#### Required Fields
+
+The following fields are **required** to be set in, otherwise they won't be matched by sync:
+
+|Draft|Required Fields|Note|
+|---|---|---|
+| `ProductDraft` | `key` |  Also, the products in the target project are expected to have the `key` fields set. | 
+| `ProductVariantDraft`  | `key`, `sku` |  Also, all the variants in the target project are expected to have the `key` and `sku` fields set. | 
+
+#### Reference Resolution 
+
+In commercetools, a reference can be created by providing the key instead of the ID with the type [ResourceIdentifier](https://docs.commercetools.com/api/types#resourceidentifier).
+When the reference key is provided with a `ResourceIdentifier`, the sync will resolve the resource with the given key and use the ID of the found resource to create or update a reference.
+Therefore, in order for the sync to resolve the actual ids of those references, `ResourceIdentifier`s with their `key`s have to be supplied. 
+
+| Field|Type|
+|:---|:---|
+| `productType` - **Required** | ResourceIdentifier to a ProductType | 
+| `categories` | Set of ResourceIdentifier for a Category |
+| `taxCategory` | ResourceIdentifier to a TaxCategory | 
+| `state` | ResourceIdentifier for a State | Optional |
+| `variants.prices.channel` | ResourceIdentifier to a Channel | 
+| `variants.prices.customerGroup` | Reference to a CustomerGroup | 
+| `variants.prices.custom.type` | ResourceIdentifier to a Type | 
+| `variants.assets.custom.type` | ResourceIdentifier to a Type | 
+| `variants.attributes` * | only for `ReferenceType` attributes the Reference object (i.e ProductType, Category) | 
```

Comment:
Not sure what is meant here? Did you want to say that only particular attribute references are supported?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/639/commits/47cafe259b21321ded5812acf0988ec795c890c9, https://github.com/commercetools/commercetools-sync-java/pull/639/commits/85387203ad99343b17e4ef2304c911ce8d4fcc30